essence-of-live-coding/src/LiveCoding/Migrate/NoMigration.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE Arrows #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveTraversable #-}
+
+{-|
+Module      : LiveCoding.Migrate.NoMigration
+Description : Mechanism to save state in a Cell without requiring a Data instance.
+
+If a data type is wrapped in 'NoMigration' then it can be used in a 'Cell' without requiring 
+it to have a 'Data' instance. The consequence is that if the type has changed
+in between a livereload, then the previous saved value will be discarded, and no migration attempt
+will happen.
+
+'LiveCoding' does not export 'delay' and 'changes' from this module. These functions should be
+used with a qualified import.
+-}
+module LiveCoding.Migrate.NoMigration where
+
+-- base
+import Data.Data
+    ( Data(dataTypeOf, gunfold, toConstr),
+      Typeable,
+      mkConstr,
+      mkDataType,
+      Constr,
+      DataType,
+      Fixity(Prefix) )
+
+-- essence-of-live-coding
+import LiveCoding.Cell ( Cell(Cell) )
+import LiveCoding.Cell.Monad ( hoistCellKleisli )
+import Control.Arrow ( Arrow(arr), returnA, (>>>) )
+import qualified LiveCoding.Cell.Feedback as Feedback
+import Control.Monad (guard)
+
+-- |Isomorphic to @'Maybe' a@ but has a different Data instance
+data NoMigration a = Initialized a | Uninitialized
+    deriving (Show, Eq, Functor, Foldable, Traversable)
+
+fromNoMigration :: a -> NoMigration a -> a
+fromNoMigration _ (Initialized a) = a
+fromNoMigration a Uninitialized = a
+
+dataTypeNoMigration :: DataType
+dataTypeNoMigration = mkDataType "NoMigration" [initializedConstr, uninitializedConstr]
+
+initializedConstr :: Constr
+initializedConstr = mkConstr dataTypeNoMigration "Initialized" [] Prefix
+
+uninitializedConstr :: Constr
+uninitializedConstr = mkConstr dataTypeNoMigration "Uninitialized" [] Prefix
+
+-- |This 'Data' instance is such that deserialization always returns an Uninitialized value.
+instance (Typeable a) => Data (NoMigration a) where
+  dataTypeOf _ = dataTypeNoMigration
+  toConstr (Initialized _) = initializedConstr
+  toConstr Uninitialized = uninitializedConstr
+  gunfold _cons nil _ = nil Uninitialized
+
+-- Should be identical to traverse'
+resampleNoMigrate :: Monad m => Cell m a b -> Cell m (NoMigration a) (NoMigration b)
+resampleNoMigrate = hoistCellKleisli morph
+  where
+    morph _ s Uninitialized = return (Uninitialized, s)
+    morph singleStep s (Initialized a) = do
+      (!b , s' ) <- singleStep s a
```

Comment:
Why is `b` forced? Do you want to make sure `singleStep` is guaranteed to be evaluated? Then maybe `!(b, s')` will do what you want?

Reply:
This is was basically copy paste from here: https://github.com/turion/essence-of-live-coding/blob/30eb53698242288922ef5601fab69f87b08fc00b/essence-of-live-coding/src/LiveCoding/Cell/Resample.hs#L31-L37 .

Reply:
Hmm, curious. In that case it would be better to rewrite it like [`resampleMaybe`](https://github.com/turion/essence-of-live-coding/blob/30eb53698242288922ef5601fab69f87b08fc00b/essence-of-live-coding/src/LiveCoding/Cell/Resample.hs#L39).